### PR TITLE
popen: add runtime checks for opts field

### LINKED ISSUE #4923: popen: protect popen.opts.<...> constants from mistyping
Tarantool version: 2.5.0-2-g098324556.

The popen module exposes the list of constants that it accepts and returns:

```yml
tarantool> popen = require('popen')
tarantool> popen
---
- new: 'function: 0x40ee1a90'
  shell: 'function: 0x40ee1ab8'
  state:
    ALIVE: alive
    EXITED: exited
    SIGNALED: signaled
  signal:
    SIGTERM: 15
    SIGKILL: 9
    <...>
  stream:
    CLOSED: closed
    OPEN: open
  opts:
    PIPE: pipe
    DEVNULL: devnull
    INHERIT: inherit
    CLOSE: close
```

Mistyping like `popen.new(<...>, {stdin = popen.opts.DEVNUL})` will pass `{stdin = nil}` table and the default behaviour (inheriting) will be silently chosen.

We can set a metatable on `popen.opts` or even on all those tables including the module table itself. This metatable will define `__index` metamethod that will give an error on attempt to access a non-existing field.

I prefer to use constants from `popen.opts.<...>` as option values, because it visually distinguish a value that is a kind of enum value from an arbitrary string payload. Mistyping of, say, `'devnull'` will give an error at `popen.new()` and it would be good to receive an error for mistyping of a `popen.opts` constant.

### PR DESCRIPTION
**Pull request by artur-barsegyan**
A user can make a typo during type a popen option, for example:
popen.new(<...>, {stdin=popen.opts.DEVNU}). This patch prevents it by
__index method that checks the user's option and throw an exception in case
of error.

Closes #4923

### CODE REVIEW

**Comment by cyrillos** [comment]
The fix looks reasonable, but I left to to @Totktonada as he is the main developer of the popen modue. Also please keep open brace on the same line of `if` statement.

File: src/lua/popen.c [hunk 1]
```diff
@@ -2529,6 +2554,12 @@ tarantool_lua_popen_init(struct lua_State *L)
 		lua_pushstring(L, popen_lua_actions[i].value);
 		lua_setfield(L, -2, popen_lua_actions[i].name);
 	}
+	lua_newtable(L);
+	{
```
Comment (cyrillos) [inline src/lua/popen.c:2558]: Why this `{` block is needed?

Comment (artur-barsegyan) [reply src/lua/popen.c:2558]: That block is a result of the inspiration of other similar modules with Lua C API in Tarantool.

**Comment by artur-barsegyan** [comment]
This PR — it's my first try to check the contribution path.

Actually, I didn't sure that PR is needed for the master branch because this behavior isn't consistent with another Tarantool modules and APIs.

**Comment by cyrillos** [comment]
After more precise review:
1) We need a test for validation
2) The `opt` table should be traversing `popen_lua_actions` table for keys `stdin`, `stdout`, `stderr` only because there might be other options as well (if only I'm not missing something obvious).

That said at least without a test case the patch is unacceptable, sorry. Please consider either create a new patch or update this one.

**Comment by kyukhin** [comment]
No response for more than month.

**Pull request closed without merging**